File: app/src/main/java/dev/ragnarok/fenrir/domain/IBlacklistRepository.java
package dev.ragnarok.fenrir.domain;

import dev.ragnarok.fenrir.model.User;
import dev.ragnarok.fenrir.util.Pair;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Observable;

public interface IBlacklistRepository {
    Completable fireAdd(int accountId, User user);

    Completable fireRemove(int accountId, int userId);

    Observable<Pair<Integer, User>> observeAdding();

    Observable<Pair<Integer, Integer>> observeRemoving();
}

File: app/src/main/java/dev/ragnarok/fenrir/domain/IBlacklistRepository.kt
package dev.ragnarok.fenrir.domain

import dev.ragnarok.fenrir.model.User
import dev.ragnarok.fenrir.util.Pair
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Observable

interface IBlacklistRepository {
    fun fireAdd(accountId: Int, user: User): Completable
    fun fireRemove(accountId: Int, userId: Int): Completable
    fun observeAdding(): Observable<Pair<Int, User>>
    fun observeRemoving(): Observable<Pair<Int, Int>>
}